refactor: optimize wsman repository code

amarnath-ac opened the pull request:
- Use reflect to extract data from CIM objects
- Consolidate property extraction with unified approach
- Refactor status extraction code
- Updated unit test cases for system by ID with 100% coverage.

Computer system output:
curl -k -u "standalone:G@ppm0ym" http://localhost:8181/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163 | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   820  100   820    0     0    170      0  0:00:04  0:00:04 --:--:--   170
{
  "@odata.context": "/redfish/v1/$metadata#ComputerSystem.ComputerSystem",
  "@odata.id": "/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163",
  "@odata.type": "#ComputerSystem.v1_26_0.ComputerSystem",
  "Actions": {
    "#ComputerSystem.Reset": {
      "target": "/redfish/v1/Systems/a72605c3-7494-4c63-855b-273410b4e163/Actions/ComputerSystem.Reset",
      "title": "Reset"
    }
  },
  "BiosVersion": "WSADLV57.0096.2025.0115.1007",
  "HostName": null,
  "Id": "a72605c3-7494-4c63-855b-273410b4e163",
  "Manufacturer": "Intel(R) Client Systems",
  "Model": "NUC12WSKv5",
  "Name": "a72605c3-7494-4c63-855b-273410b4e163",
  "PowerState": "On",
  "ProcessorSummary": {
    "CoreCount": null,
    "Count": 1,
    "LogicalProcessorCount": null,
    "Model": "12th Gen Intel(R) Core(TM) i5-1250P",
    "Status": {
      "Health": "OK",
      "HealthRollup": "OK",
      "State": "Enabled"
    }
  },
  "SerialNumber": "BTWS22800956",
  "SystemType": "Physical"
}

amarnath-ac committed:
refactor: optimize wsman repository code

- Use reflect to extract data from CIM objects
- Consolidate property extraction with unified approach
- Refactor status extraction code
- Updated unit test cases for system by ID.

Signed-off-by: C, Amarnath <amarnath.c@intel.com>

sudhir-intc commented:
@amarnath-ac : can you resolve the conflicts and sync with the main as this will help to review the actual changes.

amarnath-ac commented:
> @amarnath-ac : can you resolve the conflicts and sync with the main as this will help to review the actual changes.

Rebased and fixed merge conflicts

sudhir-intc reviewed (commented):
Changes LGTM.
Could you please also work towards add UT for this code.

sudhir-intc commented:
@amarnath-ac : Code changes look good. Verified the functionality too. Before we merge could you please add the UT for the code which has been added.

amarnath-ac commented:
> @amarnath-ac : Code changes look good. Verified the functionality too. Before we merge could you please add the UT for the code which has been added.

Added unit test in system_test.go to achive 100% coverage for system by endpoint.

sudhir-intc committed:
Merge branch 'main' into rf_system_enhance_optimize

sudhir-intc committed:
docs: updated the function header comment
